foxsec_dynamo_worker.py

```diff
@@ -0,0 +1,314 @@
+
+
+def dynamodb_delete_items(items):
+    """
+    Remove item ids from dynamodb
+    """
+
+    print("Removing %d items from DynamoDB" % (len(items)))
+    for item in items:
+        try:
+            DYNAMODB.delete_item(Key={'id': item})
+        except ClientError as err:
+            print("dynamodb delete_item failed: %s" % err)
+
+
+def slack_log_expiration(source_address, expires_at, blocked_at, summary, slack_messages):
+    """
+    Post a message to Slack when a WAF entry expires
+    """
+
+    search_pattern = source_address.split('/')[0]
+    escaped_search_pattern = re.escape(search_pattern)
+    slack_data = {
+        "attachments": [{
+            "fallback": "WAF Blacklist entry removed for {}".format(source_address),
+            "color": "#36a64f",
+            "pretext": "WAF Blacklist entry removed",
+            "fields": [{
+                "title": "Address",
+                "value": source_address,
+                "short": True,
+            }, {
+                "title": "Blocked",
+                "value": blocked_at,
+                "short": True
+            }, {
+                "title": "Expired",
+                "value": expires_at,
+                "short": True
+            }, {
+                "title": "Summary",
+                "value": summary
+            }],
+            "actions": [{
+                "type": "button",
+                "text": "View logs",
+                "url": "https://console.aws.amazon.com/cloudwatch/home?region=us-east-1#logs-insights:queryDetail=~(end~0~start~-3600~timeType~'RELATIVE~unit~'seconds~editorString~'fields*20*40message*20*7c*20filter*20*40message*20like*20*2f{}*2f*0a*7c*20sort*20*40timestamp*20desc*0a*7c*20limit*202000~isLiveTail~false~queryId~'webserver-prod*2fapache_access~source~'webserver-prod*2fapache_access)".format(escaped_search_pattern)
+            }],
+            "footer": "foxsec_dynamo_worker"
+        }]
+    }
+
+    # Queue message
+    slack_messages.append(slack_data)
+
+
+def slack_log_untracked(source_address, slack_messages):
+    """
+    Post a message to Slack when a WAF entry is not tracked in Dynamo
+    """
+
+    slack_data = {
+        "attachments": [{
+            "fallback": "Untracked WAF blacklist entry {} removed".format(source_address),
+            "color": "#ff004f",
+            "pretext": "Untracked WAF blacklist entry removed",
+            "fields": [{
+                "title": "Address",
+                "value": source_address,
+                "short": False
+            }],
+            "footer": "Foxsec dynamo worker"
+        }]
+    }
+
+    # Queue message
+    slack_messages.append(slack_data)
+
+
+def post_slack_messages(slack_messages):
+    """
+    Post a message to Slack
+    """
+
+    for slack_data in slack_messages:
+        # Post the message to webhook
+        response = requests.post(SLACK_WEBHOOK_URL, json=slack_data,
+                                 headers={'Content-Type': 'application/json'})
```

Comment:
Code is mixing `'` and `"`. Also this should probably either be on one line or split over three, instead of having two arguments on one line and a third on the second.